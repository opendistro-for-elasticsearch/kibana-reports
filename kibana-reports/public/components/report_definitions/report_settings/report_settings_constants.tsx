/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const REPORT_SOURCE_RADIOS = [
  {
    id: 'dashboardReportSource',
    label: 'Dashboard',
  },
  {
    id: 'visualizationReportSource',
    label: 'Visualization',
  },
  {
    id: 'savedSearchReportSource',
    label: 'Saved search',
  },
];

export const PDF_PNG_FILE_FORMAT_OPTIONS = [
  {
    id: 'pdf',
    label: 'PDF',
  },
  {
    id: 'png',
    label: 'PNG',
  },
];

export const SAVED_SEARCH_FORMAT_OPTIONS = [
  {
    id: 'csvFormat',
    label: 'CSV',
  },
  {
    id: 'xlsFormat',
    label: 'XLS',
  },
];

export const HEADER_FOOTER_CHECKBOX = [
  {
    id: 'header',
    label: 'Add header',
  },
  {
    id: 'footer',
    label: 'Add footer',
  },
];
export const REPORT_SOURCE_TYPES = {
  dashboard: 'Dashboard',
  visualization: 'Visualization',
  savedSearch: 'Saved search',
};

export const commonTimeRanges = [
  {
    start: 'now/d',
    end: 'now',
    label: 'Today so far'
  },
  {
    start: 'now/w',
    end: 'now',
    label: 'Week to date'
  },
  {
    start: 'now/M',
    end: 'now',
    label: 'Month to date'
  },
  {
    start: 'now/y',
    end: 'now',
    label: 'Year to date'
  }
]