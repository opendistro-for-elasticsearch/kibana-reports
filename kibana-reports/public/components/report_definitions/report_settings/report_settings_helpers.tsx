/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const parseInContextUrl = (url: string, parameter: string) => {
  const info = url.split('?');
  if (parameter === 'id') {
    return info[1].substring(info[1].indexOf(':') + 1, info[1].length);
  } else if (parameter === 'timeFrom') {
    return info[2].substring(info[2].indexOf('=') + 1, info[2].length);
  } else if (parameter === 'timeTo') {
    return info[3].substring(info[3].indexOf('=') + 1, info[3].length);
  }
  return 'error: invalid parameter';
};

export const getDashboardBaseUrlCreate = (
  edit: boolean,
  editDefinitionId: string,
  fromInContext: boolean
) => {
  let baseUrl;
  if (!fromInContext) {
    baseUrl = location.pathname + location.hash;
  } else {
    baseUrl = '/app/dashboards#/view/';
  }
  if (edit) {
    return baseUrl.replace(
      `opendistro_kibana_reports#/edit/${editDefinitionId}`,
      'dashboards#/view/'
    );
  } else if (fromInContext) {
    return baseUrl;
  }
  return baseUrl.replace(
    'opendistro_kibana_reports#/create',
    'dashboards#/view/'
  );
};

export const getVisualizationBaseUrlCreate = (
  edit: boolean,
  editDefinitionId: string,
  fromInContext: boolean
) => {
  let baseUrl;
  if (!fromInContext) {
    baseUrl = location.pathname + location.hash;
  } else {
    baseUrl = '/app/visualize#/edit/';
  }
  if (edit) {
    return baseUrl.replace(
      `opendistro_kibana_reports#/edit/${editDefinitionId}`,
      'visualize#/edit/'
    );
  } else if (fromInContext) {
    return baseUrl;
  }
  return baseUrl.replace(
    'opendistro_kibana_reports#/create',
    'visualize#/edit/'
  );
};

export const getSavedSearchBaseUrlCreate = (
  edit: boolean,
  editDefinitionId: string,
  fromInContext: boolean
) => {
  let baseUrl;
  if (!fromInContext) {
    baseUrl = location.pathname + location.hash;
  } else {
    baseUrl = '/app/discover#/view/';
  }
  if (edit) {
    return baseUrl.replace(
      `opendistro_kibana_reports#/edit/${editDefinitionId}`,
      'discover#/view/'
    );
  } else if (fromInContext) {
    return baseUrl;
  }
  return baseUrl.replace(
    'opendistro_kibana_reports#/create',
    'discover#/view/'
  );
};

export const getDashboardOptions = (data) => {
  let index;
  let dashboard_options = [];
  for (index = 0; index < data.length; ++index) {
    let entry = {
      value: data[index]['_id'].substring(10),
      text: data[index]['_source']['dashboard']['title'],
    };
    dashboard_options.push(entry);
  }
  return dashboard_options;
};

export const getVisualizationOptions = (data: string | any[]) => {
  let index;
  let options = [];
  for (index = 0; index < data.length; ++index) {
    let entry = {
      value: data[index]['_id'].substring(14),
      text: data[index]['_source']['visualization']['title'],
    };
    options.push(entry);
  }
  return options;
};

export const getSavedSearchOptions = (data: string | any[]) => {
  let index;
  let options = [];
  for (index = 0; index < data.length; ++index) {
    let entry = {
      value: data[index]['_id'].substring(7),
      text: data[index]['_source']['search']['title'],
    };
    options.push(entry);
  }
  return options;
};

export const handleDataToVisualReportSourceChange = (
  reportDefinitionRequest
) => {
  delete reportDefinitionRequest.report_params.core_params.saved_search_id;
  delete reportDefinitionRequest.report_params.core_params.limit;
  delete reportDefinitionRequest.report_params.core_params.excel;
  reportDefinitionRequest.report_params.core_params.report_format = 'pdf';
};
