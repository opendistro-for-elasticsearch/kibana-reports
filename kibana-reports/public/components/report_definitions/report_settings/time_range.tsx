/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import moment from 'moment';
import React, { useState, useEffect } from 'react';
import { parseInContextUrl } from './report_settings_helpers';
import dateMath from '@elastic/datemath';
import {
  EuiFormRow,
  EuiGlobalToastList,
  EuiSuperDatePicker,
} from '@elastic/eui';
import { commonTimeRanges } from './report_settings_constants';

export function TimeRangeSelect(props) {
  const {
    reportDefinitionRequest,
    timeRange,
    edit,
    id,
    httpClientProps,
    showTimeRangeError,
  } = props;

  const [recentlyUsedRanges, setRecentlyUsedRanges] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [start, setStart] = useState('now-30m');
  const [end, setEnd] = useState('now');

  const [toasts, setToasts] = useState([]);

  const addInvalidTimeRangeToastHandler = () => {
    const errorToast = {
      title: 'Invalid time range selected',
      color: 'danger',
      iconType: 'alert',
      id: 'timeRangeErrorToast',
    };
    setToasts(toasts.concat(errorToast));
  };

  const handleInvalidTimeRangeToast = () => {
    addInvalidTimeRangeToastHandler();
  };

  const removeToast = (removedToast) => {
    setToasts(toasts.filter((toast) => toast.id !== removedToast.id));
  };

  const isValidTimeRange = (
    timeRangeMoment: number | moment.Moment,
    limit: string,
    handleInvalidTimeRangeToast: any
  ) => {
    if (limit === 'start') {
      if (!timeRangeMoment || !timeRangeMoment.isValid()) {
        handleInvalidTimeRangeToast();
      }
    } else if (limit === 'end') {
      if (
        !timeRangeMoment ||
        !timeRangeMoment.isValid() ||
        timeRangeMoment > moment.now()
      ) {
        handleInvalidTimeRangeToast();
      }
    }
  };

  const setDefaultEditTimeRange = (duration, unmounted) => {
    let time_difference = moment.now() - duration;
    const fromDate = new Date(time_difference);
    parseTimeRange(fromDate, end, reportDefinitionRequest);
    if (!unmounted) {
      setStart(fromDate.toISOString());
      setEnd(end);
    }
  };

  // valid time range check for absolute time end date
  const checkValidAbsoluteEndDate = (end) => {
    let endDate = new Date(end);
    let nowDate = new Date(moment.now());
    let valid = true;
    if (endDate.getTime() > nowDate.getTime()) {
      end = 'now';
      valid = false;
    }
    return valid;
  };

  useEffect(() => {
    let unmounted = false;
    // if we are coming from the in-context menu
    if (window.location.href.indexOf('?') > -1) {
      const url = window.location.href;
      const timeFrom = parseInContextUrl(url, 'timeFrom');
      const timeTo = parseInContextUrl(url, 'timeTo');
      parseTimeRange(timeFrom, timeTo, reportDefinitionRequest);
      if (!unmounted) {
        setStart(timeFrom);
        setEnd(timeTo);
      }
    } else {
      if (edit) {
        httpClientProps
          .get(`../api/reporting/reportDefinitions/${id}`)
          .then(async (response: {}) => {
            let duration =
              response.report_definition.report_params.core_params
                .time_duration;
            duration = moment.duration(duration);
            setDefaultEditTimeRange(duration, unmounted);
          })
          .catch((error) => {
            console.error(
              'error in fetching report definition details:',
              error
            );
          });
      } else {
        parseTimeRange(start, end, reportDefinitionRequest);
      }
    }
    return () => {
      unmounted = true;
    };
  }, []);

  const onTimeChange = ({ start, end }) => {
    isValidTimeRange(
      dateMath.parse(start),
      'start',
      handleInvalidTimeRangeToast
    );
    isValidTimeRange(
      dateMath.parse(end, { roundUp: true }),
      'end',
      handleInvalidTimeRangeToast
    );

    const recentlyUsedRange = recentlyUsedRanges.filter((recentlyUsedRange) => {
      const isDuplicate =
        recentlyUsedRange.start === start && recentlyUsedRange.end === end;
      return !isDuplicate;
    });
    const validEndDate = checkValidAbsoluteEndDate(end);
    if (!validEndDate) {
      handleInvalidTimeRangeToast();
      return;
    }

    recentlyUsedRange.unshift({ start, end });
    setStart(start);
    setEnd(end);
    setRecentlyUsedRanges(
      recentlyUsedRange.length > 10
        ? recentlyUsedRange.slice(0, 9)
        : recentlyUsedRange
    );
    setIsLoading(true);
    startLoading();
    parseTimeRange(start, end, reportDefinitionRequest);
  };

  const parseTimeRange = (start, end, reportDefinitionRequest) => {
    timeRange.timeFrom = dateMath.parse(start);
    timeRange.timeTo = dateMath.parse(end);
    const timeDuration = moment.duration(
      dateMath.parse(end).diff(dateMath.parse(start))
    );
    reportDefinitionRequest.report_params.core_params.time_duration = timeDuration.toISOString();
  };

  const startLoading = () => {
    setTimeout(stopLoading, 1000);
  };

  const stopLoading = () => {
    setIsLoading(false);
  };


  return (
    <div>
      <div>
        <EuiFormRow
          label="Time range"
          helpText="Time range is relative to the report creation date on the report trigger."
          isInvalid={showTimeRangeError}
          error={'Invalid time range selected.'}
        >
          <EuiSuperDatePicker
            isDisabled={false}
            isLoading={isLoading}
            start={start}
            end={end}
            onTimeChange={onTimeChange}
            showUpdateButton={false}
            commonlyUsedRanges={commonTimeRanges}
          />
        </EuiFormRow>
      </div>
      <div>
        <EuiGlobalToastList
          toasts={toasts}
          dismissToast={removeToast}
          toastLifeTimeMs={6000}
        />
      </div>
    </div>
  );
}
