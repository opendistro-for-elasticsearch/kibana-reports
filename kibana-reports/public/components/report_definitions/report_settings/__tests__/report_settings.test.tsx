/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ReportSettings } from '../report_settings';
import 'babel-polyfill';
import 'regenerator-runtime';
import httpClientMock from '../../../../../test/httpMockClient';
import { act } from 'react-dom/test-utils';

const emptyRequest = {
  report_params: {
    report_name: '',
    report_source: '',
    description: '',
    core_params: {
      base_url: '',
      report_format: '',
      time_duration: '',
    },
  },
  delivery: {
    delivery_type: '',
    delivery_params: {},
  },
  trigger: {
    trigger_type: '',
    trigger_params: {},
  },
  time_created: 0,
  last_updated: 0,
  status: '',
};

let timeRange = {
  timeFrom: new Date(123456789),
  timeTo: new Date(1234567890),
};

const dashboardHits = {
  hits: [
    {
      _id: 'dashboard:abcdefghijklmnop12345',
      _source: {
        dashboard: {
          description: 'mock dashboard value',
          hits: 0,
          timeFrom: 'now-24h',
          timeTo: 'now',
          title: 'Mock Dashboard',
        },
      },
    },
  ],
};

const visualizationHits = {
  hits: [
    {
      _id: 'visualization:abcdefghijklmnop12345',
      _source: {
        visualization: {
          description: 'mock visualization value',
          title: 'Mock Visualization',
        },
      },
    },
  ],
};

const savedSearchHits = {
  hits: [
    {
      _id: 'search:abcdefghijklmnop12345',
      _source: {
        search: {
          title: 'Mock saved search value',
        },
      },
    },
  ],
};

describe('<ReportSettings /> panel', () => {
  jest.spyOn(console, 'log').mockImplementation(() => {});
  test('render component', () => {
    const { container } = render(
      <ReportSettings
        edit={false}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('render edit, dashboard source', async () => {
    const promise = Promise.resolve();
    let report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Dashboard',
        description: 'test description',
        core_params: {
          base_url: 'http://localhost:5601',
          report_format: 'pdf',
          header: 'header content',
          footer: 'footer content',
          time_duration: 'PT30M',
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {},
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      hits: dashboardHits,
    });

    const { container } = render(
      <ReportSettings
        edit={true}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });

  test('render edit, visualization source', async () => {
    const promise = Promise.resolve();
    let report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Visualization',
        description: 'test description',
        core_params: {
          base_url: 'http://localhost:5601',
          report_format: 'png',
          header: 'header content',
          footer: 'footer content',
          time_duration: 'PT30M',
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {},
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      hits: visualizationHits,
    });

    const { container } = render(
      <ReportSettings
        edit={true}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });

  test('render edit, saved search source', async () => {
    const promise = Promise.resolve();
    let report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Saved search',
        description: 'test description',
        core_params: {
          base_url: 'http://localhost:5601',
          report_format: 'csv',
          header: 'test header content',
          footer: 'test footer content',
          time_duration: 'PT30M',
          saved_search_id: 'abcdefghijk',
          limit: 10000,
          excel: true,
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {},
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      hits: savedSearchHits,
    });

    const { container } = render(
      <ReportSettings
        edit={true}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });

  test('dashboard create from in-context', async () => {
    window = Object.create(window);
    Object.defineProperty(window, 'location', {
      configurable: true,
      value: {
        href:
          'http://localhost:5601/app/opendistro_kibana_reports#/create?previous=dashboard:7adfa750-4c81-11e8-b3d7-01146121b73d?timeFrom=2020-10-26T20:52:56.382Z?timeTo=2020-10-27T20:52:56.384Z',
      },
    });

    const promise = Promise.resolve();

    let report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Dashboard',
        description: '',
        core_params: {
          base_url: 'http://localhost:5601',
          report_format: 'png',
          header: '',
          footer: '',
          time_duration: 'PT30M',
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {},
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      hits: dashboardHits,
    });

    const { container } = render(
      <ReportSettings
        edit={false}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });

  test('visualization create from in-context', async () => {
    // @ts-ignore
    delete window.location; // reset window.location.href for in-context testing

    window = Object.create(window);
    Object.defineProperty(window, 'location', {
      configurable: true,
      value: {
        href:
          'http://localhost:5601/app/opendistro_kibana_reports#/create?previous=visualize:7adfa750-4c81-11e8-b3d7-01146121b73d?timeFrom=2020-10-26T20:52:56.382Z?timeTo=2020-10-27T20:52:56.384Z',
      },
    });

    const promise = Promise.resolve();

    let report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Visualization',
        description: '',
        core_params: {
          base_url: 'http://localhost:5601',
          report_format: 'pdf',
          header: '',
          footer: '',
          time_duration: 'PT30M',
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {},
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      hits: visualizationHits,
    });

    const { container } = render(
      <ReportSettings
        edit={false}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });

  test('saved search create from in-context', async () => {
    // @ts-ignore
    delete window.location; // reset window.location.href for in-context testing

    window = Object.create(window);
    Object.defineProperty(window, 'location', {
      value: {
        href:
          'http://localhost:5601/app/opendistro_kibana_reports#/create?previous=discover:7adfa750-4c81-11e8-b3d7-01146121b73d?timeFrom=2020-10-26T20:52:56.382Z?timeTo=2020-10-27T20:52:56.384Z',
      },
    });

    const promise = Promise.resolve();

    let report_definition = {
      report_params: {
        report_name: 'test create report definition trigger',
        report_source: 'Saved search',
        description: '',
        core_params: {
          base_url: 'http://localhost:5601',
          report_format: 'csv',
          header: '',
          footer: '',
          time_duration: 'PT30M',
          saved_search_id: 'abcdefghijk',
          limit: 10000,
          excel: true,
        },
      },
      delivery: {
        delivery_type: '',
        delivery_params: {},
      },
      trigger: {
        trigger_type: 'Schedule',
        trigger_params: {},
      },
    };

    httpClientMock.get = jest.fn().mockResolvedValue({
      report_definition,
      hits: savedSearchHits,
    });

    const { container } = render(
      <ReportSettings
        edit={false}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={false}
        showTimeRangeError={false}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });

  test('display errors on create', async () => {
    const promise = Promise.resolve();
    const { container } = render(
      <ReportSettings
        edit={false}
        reportDefinitionRequest={emptyRequest}
        httpClientProps={httpClientMock}
        timeRange={timeRange}
        showSettingsReportNameError={true}
        showTimeRangeError={true}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
    await act(() => promise);
  });
});
