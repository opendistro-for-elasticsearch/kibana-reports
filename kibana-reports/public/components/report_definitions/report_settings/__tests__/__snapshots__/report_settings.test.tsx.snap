// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportSettings /> panel render component 1`] = `
<div
  class="euiPanel euiPanel--paddingLarge euiPageContent"
>
  <div
    class="euiPageHeader euiPageHeader--responsive"
  >
    <h2
      class="euiTitle euiTitle--medium"
    >
      Report Settings
    </h2>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
  />
  <div
    class="euiPageContentBody"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Name
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-describedby="random_html_id-help"
                  class="euiFieldText"
                  id="random_html_id"
                  placeholder="Report name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="euiFormHelpText euiFormRow__text"
              id="random_html_id-help"
            >
              Valid characters are a-z, A-Z, 0-9, (), [], _ (underscore), - (hyphen) and (space)
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="max-width: 600px;"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Description
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-describedby="random_html_id-help"
                  class="euiFieldText"
                  id="random_html_id"
                  placeholder="Describe this report"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="euiFormHelpText euiFormRow__text"
              id="random_html_id-help"
            >
              Optional
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div
      class="euiFormRow"
      id="random_html_id-row"
    >
      <div
        class="euiFormRow__labelWrapper"
      >
        <label
          class="euiFormLabel euiFormRow__label"
          for="random_html_id"
        >
          Report source
        </label>
      </div>
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          aria-describedby="random_html_id-help"
          id="random_html_id"
        >
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              checked=""
              class="euiRadio__input"
              id="dashboardReportSource"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="dashboardReportSource"
            >
              Dashboard
            </label>
          </div>
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              class="euiRadio__input"
              id="visualizationReportSource"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="visualizationReportSource"
            >
              Visualization
            </label>
          </div>
          <div
            class="euiRadio euiRadioGroup__item"
          >
            <input
              class="euiRadio__input"
              id="savedSearchReportSource"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="savedSearchReportSource"
            >
              Saved search
            </label>
          </div>
        </div>
        <div
          class="euiFormHelpText euiFormRow__text"
          id="random_html_id-help"
        >
          Where is the report generated from
        </div>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
    <div>
      <div
        class="euiFormRow"
        id="random_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            class="euiFormLabel euiFormRow__label"
            for="random_html_id"
          >
            Select dashboard
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <select
                class="euiSelect"
                id="random_html_id"
              />
              <div
                class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div>
        <div
          class="euiFormRow"
          id="random_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Time range
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--directionRow euiSuperDatePicker__flexWrapper"
            >
              <div
                class="euiFlexItem"
              >
                <div
                  class="euiFormControlLayout euiFormControlLayout--group euiSuperDatePicker"
                >
                  <div
                    class="euiPopover euiPopover--anchorDownLeft"
                    id="QuickSelectPopover"
                  >
                    <div
                      class="euiPopover__anchor euiQuickSelectPopover__anchor"
                    >
                      <button
                        aria-label="Date quick select"
                        class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--iconRight euiFormControlLayout__prepend"
                        data-test-subj="superDatePickerToggleQuickMenuButton"
                        type="button"
                      >
                        <span
                          class="euiButtonEmpty__content"
                        >
                          <svg
                            aria-hidden="true"
                            class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonEmpty__icon"
                            focusable="false"
                            height="16"
                            role="img"
                            viewBox="0 0 16 16"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          />
                          <span
                            class="euiButtonEmpty__text euiQuickSelectPopover__buttonText"
                          >
                            <svg
                              aria-hidden="true"
                              class="euiIcon euiIcon--medium euiIcon-isLoading"
                              focusable="false"
                              height="16"
                              role="img"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            />
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="euiFormControlLayout__childrenWrapper"
                  >
                    <div
                      class="euiDatePickerRange euiDatePickerRange--inGroup"
                    >
                      <button
                        class="euiSuperDatePicker__prettyFormat"
                        data-test-subj="superDatePickerShowDatesButton"
                      >
                        Last 30 minutes
                        <span
                          class="euiSuperDatePicker__prettyFormatLink"
                        >
                          Show dates
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <span
                  class="euiToolTipAnchor"
                >
                  <button
                    class="euiButton euiButton--primary euiSuperUpdateButton euiButton--fill"
                    data-test-subj="superDatePickerApplyTimeButton"
                    type="button"
                  >
                    <span
                      class="euiButton__content"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--medium euiIcon-isLoading euiButton__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      />
                      <span
                        class="euiButton__text euiSuperUpdateButton__text"
                      >
                        Refresh
                      </span>
                    </span>
                  </button>
                </span>
              </div>
            </div>
            <div
              class="euiFormHelpText euiFormRow__text"
              id="random_html_id-help"
            >
              Time range is relative to the report creation date on the report trigger
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div>
        <div
          class="euiFormRow"
          id="random_html_id-row"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              File format
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              id="random_html_id"
            >
              <div
                class="euiRadio euiRadioGroup__item"
              >
                <input
                  checked=""
                  class="euiRadio__input"
                  id="pdfFormat"
                  type="radio"
                  value=""
                />
                <div
                  class="euiRadio__circle"
                />
                <label
                  class="euiRadio__label"
                  for="pdfFormat"
                >
                  PDF
                </label>
              </div>
              <div
                class="euiRadio euiRadioGroup__item"
              >
                <input
                  class="euiRadio__input"
                  id="pngFormat"
                  type="radio"
                  value=""
                />
                <div
                  class="euiRadio__circle"
                />
                <label
                  class="euiRadio__label"
                  for="pngFormat"
                >
                  PNG
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--l"
        />
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div>
        <div
          class="euiCheckbox"
        >
          <input
            class="euiCheckbox__input"
            id="includeHeaderCheckbox"
            type="checkbox"
          />
          <div
            class="euiCheckbox__square"
          />
          <label
            class="euiCheckbox__label"
            for="includeHeaderCheckbox"
          >
            Include header
          </label>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div>
        <div
          class="euiCheckbox"
        >
          <input
            class="euiCheckbox__input"
            id="includeFooterCheckbox"
            type="checkbox"
          />
          <div
            class="euiCheckbox__square"
          />
          <label
            class="euiCheckbox__label"
            for="includeFooterCheckbox"
          >
            Include footer
          </label>
        </div>
      </div>
    </div>
    <div
      class="euiSpacer euiSpacer--l"
    />
  </div>
</div>
`;
